### ISSUE TITLE
TypeError is thrown when you try to use keyTyped after removing keyPressed/keyReleased callback functions

### ISSUE THREAD

**Issue by Gigabyte-Giant** [Khan/live-editor#421]
To reproduce:
 * Open your Javascript console.
 * Navigate to [this program](https://www.khanacademy.org/computer-programming/notepad-wip/6665063086161920)
 * Select all of the code with `Ctrl+A` (or `Cmd+A`); and delete the code.
 * Copy and paste [this code](https://gist.github.com/Gigabyte-Giant/0de4c14a8dc1baddea35) into the program.
 * Shift browser focus to the program (by clicking on the canvas), and press a key on your keyboard.
 * **Observe the error**.

I assume the scratchpad is still trying to invoke the (previously user-defined) callbacks `keyPressed` and `keyReleased`?

Though I haven't tested yet, I assume the same thing will occur with `mouseClicked` (*if* the original program had defined callback functions for `mousePressed` and `mouseReleased`).

**Comment by Gigabyte-Giant** [comment]
Just ran a quick test; this **does not** occur for `mousePressed`/`mouseReleased`/`mouseClicked`...

**Comment by JavascriptFTW** [comment]
This behavior also works in reverse, i.e. add `keyTyped` then remove and add `keyPressed` and `keyReleased`.

**Comment by kevinbarabash** [comment]
Hmm... we should probably redefine `mousePressed` et al to be dummy functions if they've been defined once and then removed.

**Comment by Gigabyte-Giant** [comment]
@kevinbarabash, I agree, but it would appear that you're already doing something like this for the `mouse` family of functions (hence me originally stating that this *doesn't* occur with `mouse*`).

**Comment by kevinbarabash** [comment]
@Gigabyte-Giant the link to the program doesn't work anymore.  Can you post a working link?

**Comment by Gigabyte-Giant** [comment]
[Here](https://www.khanacademy.org/computer-programming/new-program/3382983689) @kevinbarabash. To replicate the issue, remove the `keyPressed` and `keyReleased` functions, and insert a `keyTyped` function instead.

**Comment by kevinbarabash** [comment]
Thanks.
